auth_server/src/realms.rs

```diff
@@ -61,6 +65,39 @@ pub async fn receive_realm_pings(auth_db: std::sync::Arc<AuthDatabase>) -> Resul
     }
 }
 
+async fn get_realm_list(auth_database: std::sync::Arc<AuthDatabase>, account_id: u32) -> Result<Vec<Realm>> {
+    //TODO(wmxd): it will be good idea to cache the database stuff
+    //TODO(wmxd): for now it will be better select realms and number_of_chars in one database trip (eg: left join)
+    let db_realms = auth_database.get_all_realms().await?;
+    let mut realms = Vec::with_capacity(db_realms.len());
+    for realm in db_realms {
+        let num_characters = auth_database.get_num_characters_on_realm(account_id, realm.id).await?;
+
+        // TODO: Use flags from DB.
+        let mut flag = Realm_RealmFlag::empty();
```

Comment:
`Realm_RealmFlag` doesn't have a `from` impl for converting from the inner u8 yet. If it did, this whole chunk of code could be much simpler.

Reply:
I usually enter a GH issue with a feature request on the wowm repo and then put a link in the comment like "TODO refactor this when this GH issue is fixed"